### DISCUSSION TITLE
Github App Rate Limit using client_id and client_secret

### DISCUSSION THREAD

**Discussion by wei** [community/community#24482] [Apps, API and Webhooks]
<p>Hi there,</p>
<p>According to: &lt;<a href="https://developer.github.com/v3/#increasing-the-unauthenticated-rate-limit-for-oauth-applications" rel="nofollow noopener">https://developer.github.com/v3/#increasing-the-unauthenticated-rate-limit-for-oauth-applications</a>&gt; client_id and client_secret can be used as part of the query string to increase the rate limit for OAuth applications. </p>
<p>However, this does not work for Github Apps. Using the client_id and client_secret from Github App “Settings” &gt; “OAuth credentials” does not get an increased rate limit. The parameters are simply ignored.</p>
<p>Please help.</p>
<p>Thanks!</p>
<p>Wei</p>

**Comment by wei** [comment]
<p>Got a response from Ivan from Github. Basically, this is working as intended for now. For the time being, the suggested workaround is to create another OAuth Application to make unauthenticated calls.</p>

**Comment by lee-dohm** [comment]
<p>GitHub Apps and OAuth Apps use different security models and rate limiting methods. See <a href="https://developer.github.com/apps/differences-between-apps/" rel="nofollow noopener">the documentation on the differences between GitHub Apps and OAuth Apps</a> and <a href="https://developer.github.com/apps/building-github-apps/understanding-rate-limits-for-github-apps/" rel="nofollow noopener">rate limiting for GitHub Apps</a> for details.</p>
<p>I hope that helps!</p>
